src/Workspaces/Core/Portable/Serialization/SerializerService_Reference.cs

```diff
@@ -515,8 +505,10 @@ private static void GetMetadata(Stream stream, long length, out ModuleMetadata m
             pinnedObject = new PinnedObject(array);
         }
 
-        metadata = ModuleMetadata.CreateFromMetadata(pinnedObject.GetPointer(), (int)length);
-        lifeTimeObject = pinnedObject;
+        // PinnedObject will be kept alive as long as the ModuleMetadata is alive due to passing its .Dispose method in
+        // as the onDispose callback of the metadata.
+        return ModuleMetadata.CreateFromMetadata(
+            pinnedObject.GetPointer(), (int)length, pinnedObject.Dispose);
```

Comment:
@ToddGrun i thought thsi was correct, but it may be a change in behavior from what we used to do.  specificlaly, in the old code we created the metadaa from the pinned obejct, and added the pinned object back to put into the lifetime-map (https://github.com/dotnet/roslyn/pull/73112/files#diff-d326cee48834d32e28eaa3be997213b36242745d1ada523800959e563088b305R426).    But then nothing ever actually disposed those pinned objects.  

So we've subtly changed behavior here which could release data on the host side, which could explain the error being seen.